mala/network/trainer.py

```diff
@@ -524,37 +533,26 @@ def __prepare_to_train(self, optimizer_dict):
             if self.data.parameters.use_lazy_loading:
                 do_shuffle = False
 
-            self.train_sampler = torch.utils.data.\
-                distributed.DistributedSampler(self.data.training_data_sets[0],
-                                               num_replicas=hvd.size(),
-                                               rank=hvd.rank(),
-                                               shuffle=do_shuffle)
-
-            self.validation_sampler = torch.utils.data.\
-                distributed.DistributedSampler(self.data.validation_data_sets[0],
-                                               num_replicas=hvd.size(),
-                                               rank=hvd.rank(),
-                                               shuffle=False)
-
-            if self.data.test_data_sets:
-                self.test_sampler = torch.utils.data.\
-                    distributed.DistributedSampler(self.data.test_data_sets[0],
-                                                   num_replicas=hvd.size(),
-                                                   rank=hvd.rank(),
+            if self.parameters_full.use_distributed_sampler_train:
+                self.train_sampler = torch.utils.data.\
+                    distributed.DistributedSampler(self.data.training_data_sets,
```

Comment:
```suggestion
                    distributed.DistributedSampler(self.data.training_data_sets[0],
```
The sampler needs to be associated with an existing dataset. The existing line is trying to associate a sampler with the list of data sets, which does not work.

Setting up the sampler for `[0]` works for the in-memory loader only. For the lazy-loading case where there are actually multiple datasets in the `self.data.training_data_sets` list, I think we will need to create and manage an individual sampler _per_ dataset.